demos/python_demos/multi_camera_multi_person_tracking/tools/run_evaluate.py

```diff
@@ -0,0 +1,153 @@
+"""
+
+import argparse
+import json
+import logging as log
+
+import motmetrics as mm
+import numpy as np
+from lxml import etree
+from tqdm import tqdm
+
+from mc_tracker.sct import TrackedObj
+from utils.misc import set_log_config
+
+set_log_config()
+
+
+def read_gt_tracks(gt_filenames, size_divisor=1, skip_frames=0, skip_heavy_occluded_objects=False):
+    min_last_frame_idx = -1
+    camera_tracks = [[] for _ in gt_filenames]
+    for i, filename in enumerate(gt_filenames):
+        last_frame_idx = -1
+        tree = etree.parse(filename)
+        root = tree.getroot()
+        for track_xml_subtree in tqdm(root, desc='Reading ' + filename):
+            if track_xml_subtree.tag != 'track':
+                continue
+            track = {'id': None, 'boxes': [], 'timestamps': []}
+            for box_tree in track_xml_subtree.findall('box'):
+                if skip_frames > 0 and int(box_tree.get('frame')) % skip_frames == 0:
+                    continue
+                occlusion = [tag.text for tag in box_tree if tag.attrib['name'] == 'occlusion'][0]
+                if skip_heavy_occluded_objects and occlusion == 'heavy_occluded':
+                    continue
+                x_left = int(float(box_tree.get('xtl'))) // size_divisor
+                x_right = int(float(box_tree.get('xbr'))) // size_divisor
+                y_top = int(float(box_tree.get('ytl'))) // size_divisor
+                y_bottom = int(float(box_tree.get('ybr'))) // size_divisor
+                assert x_right > x_left
+                assert y_bottom > y_top
+                track['boxes'].append([x_left, y_top, x_right, y_bottom])
+                track['timestamps'].append(int(box_tree.get('frame')) // size_divisor)
+                last_frame_idx = max(last_frame_idx, track['timestamps'][-1])
+                id = [int(tag.text) for tag in box_tree if tag.attrib['name'] == 'id'][0]
+            track['id'] = id
+            camera_tracks[i].append(track)
+        if min_last_frame_idx < 0:
+            min_last_frame_idx = last_frame_idx
+        else:
+            min_last_frame_idx = min(min_last_frame_idx, last_frame_idx)
+
+    return camera_tracks, min_last_frame_idx
+
+
+def get_detections_from_tracks(tracks_history, time):
+    active_detections = [[] for _ in tracks_history]
+    for i, camera_hist in enumerate(tracks_history):
+        for track in camera_hist:
+            if time in track['timestamps']:
+                idx = track['timestamps'].index(time)
+                active_detections[i].append(TrackedObj(track['boxes'][idx], track['id']))
+    return active_detections
+
+
+def check_contain_duplicates(all_detections):
+    for detections in all_detections:
+        all_labels = [obj.label for obj in detections]
+        uniq = set(all_labels)
+        if len(all_labels) != len(uniq):
+            return True
+
+    return False
+
+
+def main():
+    """Computes MOT metrics for the person recognition demo"""
```

Comment:
It seems the tools don't clarify where to get arguments for `--gt_files`. The tools look to be for a narrow circle of users who know what all this is about. Of course it is possible to dig into and find that this is about `motmetrics` and so on, but are users supposed to do this?

What is the difference between this tool and `run_history_visualize.py`? Both accept `--gt_files` both seem to compute the metric. Is it possible to merge the tools?

Reply:
`run_evaluate.py` evaluates metrics for multi camera multi person tracker. Here you must have annotated data to compare your results with ground truth. This demo show online IDs for every bounding box and all bounding boxes are merged to tracks according to the ID. Our algorithm merges tracks by some criterion and the final result may be different from the  showed online. So, to visualize this final result we use `run_history_visualize.py` script. Also here is an option to show your results and ground truth together to compare it visually. Yes, these two scripts have common functions but I would not like to merge them. They have different purpose although both show quality of the algorithm. What about `--gt_files`, it is possible to obtain them only via annotation, we can't provide it. We use here annotation from CVAT.

Reply:
Will you consider adding your clarification above, so other people at least had a chance to figure out how to use it?

Reply:
I have added description in readme for these two scripts with format of the annotation for the ground truth files.

Reply:
Thank you